packages/i18n/src/index.ts

```diff
@@ -241,41 +245,49 @@ export type NullableTranslator<T extends BaseRecordDict, O = string> = <K extend
  * t("food.meat") // => "meat"
  * ```
  */
-export function translator<T extends BaseRecordDict, O = string>(
-  dict: () => T,
-  resolveTemplate?: TemplateResolver<O>,
-): Translator<T, O>;
-export function translator<T extends BaseRecordDict, O = string>(
-  dict: () => T | undefined,
-  resolveTemplate?: TemplateResolver<O>,
-): NullableTranslator<T, O>;
-export function translator<T extends BaseRecordDict>(
-  dict: () => T | undefined,
-  resolveTemplate: TemplateResolver = identityResolveTemplate,
-): any {
-  return (path: string, ...args: any[]) => {
-    if (path[0] === ".") path = path.slice(1);
-
+export function translator<Dict extends BaseRecordDict, Out = DefaultResolved, Args = DefaultResolveArgs>(
+  dict: () => Dict,
+  resolveTemplate?: TemplateResolver<Out, Args>,
+): Translator<Dict, Out, Args>;
+export function translator<Dict extends BaseRecordDict, Out = DefaultResolved, Args = DefaultResolveArgs>(
+  dict: () => Dict | undefined,
+  resolveTemplate?: TemplateResolver<Out, Args>,
+): NullableTranslator<Dict, Out, Args>;
+export function translator<Dict extends BaseRecordDict, Out = DefaultResolved, Args = DefaultResolveArgs>(
+  dict: () => Dict | undefined,
+  resolveTemplate: TemplateResolver<Out, Args> = identityResolveTemplate,
+): Translator<Dict, Out, Args> {
+  return (path, ...args) => {
     const value = dict()?.[path];
-
     switch (typeof value) {
       case "function":
         return value(...args);
       case "string":
-        return resolveTemplate(value, args[0]);
+        return resolveTemplate(value, ...args);
```

Comment:
Why? template args can only be a single object

Reply:
You're right, it was just the easiest way to make the types work. I have reverted it and annotated it with ts-expect-error, because casting this would be really messy